src/effgen.ml

```diff
@@ -147,6 +149,32 @@ module StaticGenerators = struct
       frequency [ (10, small_int); (5, int); (1, oneofl [ min_int; -1; 0; 1; max_int ]) ])
   ;;
 
+  let int32_gen =
+    Gen.(
+      frequency [ (4, ui32); (1, oneofl [ Int32.min_int; -1l; 0l; 1l; Int32.max_int ]) ])
+  ;;
+
+  let int64_gen =
+    Gen.(
+      frequency [ (4, ui64); (1, oneofl [ Int64.min_int; -1L; 0L; 1L; Int64.max_int ]) ])
+  ;;
+
+  (* TODO: Send a PR to QCheck and replace the fn below *)
+  let nativeint st =
+    let pos_ni = Random.State.nativeint st Nativeint.max_int in
+    match Gen.oneofl [ `Pos; `Neg ] st with
+    | `Pos -> pos_ni
+    | `Neg -> Nativeint.neg pos_ni
```

Comment:
Note: there is one more negative number than positive in a 2-complement representation:
```ocaml
utop # Nativeint.max_int;;
- : nativeint = 9223372036854775807n
utop # Nativeint.neg Nativeint.max_int;;
- : nativeint = -9223372036854775807n
utop # Nativeint.min_int;;
- : nativeint = -9223372036854775808n
```
 The above `nativeint` generator has zero chance of generating `Nativeint.max_int`, `Nativeint.min_int` and `Nativeint.neg Nativeint.max_int`.

Reply:
This is true; however, there is `nativeint_gen` function just below that returns one of the following `[ Nativeint.min_int; -1n; 0n; 1n; Nativeint.max_int ]` or a native int generated by the `nativeint` function under discussion. 

In the current implementation, `Nativeint.min_int + 1` is not generated (for keeping code simpler), but it shouldn't be critical for our purposes. This number is hard to be generated automatically by the generator lib that we use, but can be generated by manually putting its probability `1/(# of nativeints)`.